runtime/lua/vim/filetype/detect.lua

```diff
@@ -1279,22 +1278,21 @@ function M.tex(path, bufnr)
           local lpat_match, cpat_match =
             matchregex(line, [[\c^\s*\\\%(]] .. lpat .. [[\)\|^\s*\\\(]] .. cpat .. [[\)]])
           if lpat_match then
-            return 'tex'
+            format = 'tex'
           elseif cpat_match then
-            return 'context'
+            format = 'context'
           end
         end
       end
     end
-    -- TODO: add AMSTeX, RevTex, others?
-    if not vim.g.tex_flavor or vim.g.tex_flavor == 'plain' then
-      return 'plaintex'
-    elseif vim.g.tex_flavor == 'context' then
-      return 'context'
-    else
-      -- Probably LaTeX
-      return 'tex'
-    end
+  end -- if matched
+
+  -- Translation from formats to file types.  TODO:  add AMSTeX, RevTex, others?
+  if format == 'plaintex' or format == 'context' then
```

Comment:
This changes the behavior of `g:tex_flavor = 'plain'`. With `g:tex_flavor = 'plain'` filetype should be set `plaintex`, but with this PR it is set to `tex`.

Reply:
You're right, my apologies. I just pushed a fix.